rotkehlchen/api/v1/schemas.py

```diff
@@ -2083,3 +2075,103 @@ class AddressbookUpdateSchema(BaseAddressbookSchema):
 
 class AllNamesSchema(Schema):
     addresses = fields.List(EthereumAddressField(required=True), required=True)
+
+
+class SnapshotImportingSchema(Schema):
+    balances_snapshot_file = FileField(allowed_extensions=['.csv'], required=True)
+    location_data_snapshot_file = FileField(allowed_extensions=['.csv'], required=True)
+
+
+class SnapshotQuerySchema(SnapshotTimestampQuerySchema):
+    action = fields.String(
+        validate=webargs.validate.OneOf(choices=('export', 'download')),
+        load_default=None,
+    )
+    path = DirectoryField(load_default=None)
+
+    @validates_schema
+    def validate_schema(  # pylint: disable=no-self-use
+            self,
+            data: Dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        path = data.get('path')
+        action = data.get('action')
+        if action == 'export' and path is None:
+            raise ValidationError(
+                message='A path has to be provided when action is export',
+                field_name='action',
+            )
+
+
+class BalanceSnapshotSchema(Schema):
+    timestamp = TimestampField(required=True)
+    category = SerializableEnumField(enum_class=BalanceType, required=True)
+    asset_identifier = AssetField(required=True, form_with_incomplete_data=True)
+    amount = AmountField(required=True)
+    usd_value = AmountField(required=True)
+
+    @post_load
+    def make_balance(  # pylint: disable=no-self-use
+            self,
+            data: Dict[str, Any],
+            **_kwargs: Any,
+    ) -> Dict[str, Any]:
+        return {
```

Comment:
Hmmm why are you re-serializing those after validating the snapshot? This does not make sense to me.
Same as in `LocationDataSnapshotSchema`.

Is it because you only use this in the edit snapshot and the edit snapshot sends them to the same thing as CSV import? But then aren't you sanitizing the input again there?

So all these marhsmallow checks are useless?

I would say that you need to (1) have a step in the import where validated and deserialized data structures are imported in the DB and (2) from here directly inject them there.

Or am I missing something ?